scripts/check_atomic_write.py

```
@@ -0,0 +1,236 @@
+    """Heuristic: line is dominated by a docstring/string literal.
+
+    Conservative — a line that contains a forbidden pattern *only* inside a
+    string literal (e.g. an example in a docstring) should not trip the rail.
+    Detection: line, after stripping leading whitespace, starts with a quote
+    char or with the line being part of a triple-quoted block (``\"\"\"`` or
+    ``'''``). Misses some cases but cheap and false-positive-safe.
+    """
+    stripped = line.lstrip()
+    if stripped.startswith(('"""', "'''", '"', "'")):
+        return True
+    return False
```

Ruff (0.15.11)

[warning] Return the condition directly

Inline condition

(SIM103)